lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -227,6 +246,91 @@ export default function getRecipeInterface(querier: Querier, config: TypeNormali
             }
         },
 
+        assertClaims: async function (
+            this: RecipeInterface,
+            input: {
+                session: SessionContainerInterface;
+
+                claimValidators: SessionClaimValidator[];
+                userContext?: any;
+            }
+        ): Promise<void> {
+            logDebugMessage("getSession: required validator ids " + input.claimValidators.map((c) => c.id).join(", "));
+            await input.session.assertClaims(input.claimValidators, input.userContext);
+            logDebugMessage("getSession: claim assertion successful");
+        },
+
+        validateClaimsForSessionHandle: async function (
+            this: RecipeInterface,
+            input: {
+                sessionInfo: SessionInformation;
+                claimValidators: SessionClaimValidator[];
+                userContext: any;
+            }
+        ): Promise<
+            | {
+                  status: "SESSION_DOES_NOT_EXIST_ERROR";
+              }
+            | {
+                  status: "OK";
+                  invalidClaims: ClaimValidationError[];
+              }
+        > {
+            const origSessionClaimPayloadJSON = JSON.stringify(input.sessionInfo.accessTokenPayload);
+
+            let newAccessTokenPayload = await updateClaimsInPayloadIfNeeded(
+                input.sessionInfo.userId,
+                input.claimValidators,
+                input.sessionInfo.accessTokenPayload,
+                input.userContext
+            );
+
+            if (JSON.stringify(newAccessTokenPayload) !== origSessionClaimPayloadJSON) {
+                await this.mergeIntoAccessTokenPayload({
+                    accessTokenPayloadUpdate: newAccessTokenPayload,
+                    sessionHandle: input.sessionInfo.sessionHandle,
+                    userContext: input.userContext,
+                });
```

Comment:
We should check the output of this function and if it is false, we need to return SESSION_DOES_NOT_EXIST_ERROR